lib/scripting/post-process/class.ts

```diff
@@ -0,0 +1,218 @@
+		estree = ppPropertyGetDeclaration(node);
+	} else if (node.type === 'PropertyLetDeclaration') {
+		estree = ppPropertyLetDeclaration(node);
+	} else if (node.type === 'PropertySetDeclaration') {
+		estree = ppPropertySetDeclaration(node);
+	}
+	return estree;
+}
+
+function ppClassDeclaration(node: ESIToken): any {
+	let id = identifier('undefined');
+	let constructor: MethodDefinition | undefined;
+	const methodDefinitions: MethodDefinition[] = [];
+	const varStmts: Statement[] = [];
+	const ids: string[] = [];
+	for (const child of node.children) {
+		if (child.type === 'Identifier') {
+			id = child.estree;
+		} else if (child.type === 'ClassMemberDeclaration') {
+			const memberDecl = child.children[0];
+			if (memberDecl.type === 'ConstructorMemberDeclaration') {
+				constructor = memberDecl.estree;
+			} else if (memberDecl.type === 'MethodMemberDeclaration') {
+				const functionDecl = memberDecl.estree as FunctionDeclaration;
+				const functionId = functionDecl.id as Identifier;
+				methodDefinitions.push(
+					methodDefinition(functionId, 'method', functionExpression(functionDecl.body, functionDecl.params)),
+				);
+				ids.push(functionId.name);
+			} else if (memberDecl.type === 'PropertyMemberDeclaration') {
+				methodDefinitions.push(memberDecl.estree);
+			} else if (
+				memberDecl.type === 'VariableMemberDeclaration' ||
+				memberDecl.type === 'ConstantMemberDeclaration'
+			) {
+				for (const varDecl of (memberDecl.estree as VariableDeclaration).declarations) {
+					const varId = varDecl.id as Identifier;
+					varStmts.push(
+						expressionStatement(
+							assignmentExpression(
+								memberExpression(thisExpression(), varId),
+								'=',
+								varDecl.init ? varDecl.init : identifier('undefined'),
+							),
+						),
+					);
+					ids.push(varId.name);
+				}
+			}
+		}
+	}
+	if (!constructor) {
+		constructor = methodDefinition(
+			identifier('constructor'),
+			'constructor',
+			functionExpression(blockStatement([]), []),
+		);
+	}
+	let body: ClassBody = classBody([constructor, ...methodDefinitions]);
+	body = replace(body, {
+		leave: (bodyNode, parentNode) => {
+			if (bodyNode.type === 'Identifier') {
+				if (parentNode !== null && parentNode.type !== 'MethodDefinition') {
+					if (ids.includes(bodyNode.name)) {
+						if (parentNode.type === 'MemberExpression') {
+							if (parentNode.object.type === 'Identifier') {
+								if (parentNode.object.name === bodyNode.name) {
+									return memberExpression(thisExpression(), bodyNode);
+								}
+							}
+						} else {
+							return memberExpression(thisExpression(), bodyNode);
+						}
+					}
+				}
+			}
+		},
+	}) as ClassBody;
+	body.body[0].value.body.body.unshift(...varStmts);
+	return classDeclaration(id, body);
+}
+
+function ppConstructorMemberDeclaration(node: ESIToken): any {
+	let block: BlockStatement | undefined;
+	for (const child of node.children) {
+		if (child.type === 'Block') {
+			block = child.estree;
+		}
+	}
```

Comment:
This takes the *last* block, is that wanted?

Reply:
`ConstructorMemberDeclaration ::= (ConstructorModifier? WS) 'Sub' WS 'Class_Initialize' ( OpenParenthesis CloseParenthesis )? StatementTerminator Block? 'End' WS 'Sub' StatementTerminator
`

Technically there is only one block that can come ever come back. I like your idea of having a find / find children helpers.

I'd like to put some more thought into doing that right. (and all over the place)